resource/tests/__init__.py

```diff
@@ -1,22 +1,63 @@
-import json
 import os
-from typing import Optional
-import time
+
+from datetime import datetime
 from urllib.parse import quote
 
-from cryptography import x509
+from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.hazmat.backends import default_backend
+from cryptography.x509.oid import NameOID
+from cryptography import x509
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives import hashes
+
+from api import certificate_extensions
 
 ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
 
-def cert_response(cert_file=None, urlencoded=False):
-    if cert_file is None:
-        raise Exception("No certificate file specified")
-    with open(f"{ROOT_DIR}/fixtures/{cert_file}", "rb") as cert_content:
-        cert_data = cert_content.read()
-    if urlencoded:
-        cert = quote(cert_data.decode())
-    else:
-        cert = x509.load_pem_x509_certificate(cert_data, default_backend())
-    return cert
 
+def client_certificate(roles: list[str] | None = None) -> str:
+    # Generate private key
+    private_key = rsa.generate_private_key(
+        public_exponent=65537, key_size=2048, backend=default_backend()
+    )
+
+    # Define certificate details
+    subject = issuer = x509.Name(
+        [
+            x509.NameAttribute(NameOID.COUNTRY_NAME, "GB"),
+            x509.NameAttribute(NameOID.STATE_OR_PROVINCE_NAME, "London"),
+            x509.NameAttribute(
+                NameOID.ORGANIZATION_NAME, "Energy Sector Trust Framework"
+            ),
+            x509.NameAttribute(
+                NameOID.COMMON_NAME, "Energy Sector Trust Framework Client Issuer"
+            ),
+        ]
+    )
+
```

Comment:
Altered the test code to use a certificate generated on demand rather than test fixtures